sensor/LocalDomainSensor.js

```diff
@@ -0,0 +1,62 @@
+/*    Copyright 2020 Firewalla INC 
+ *
+ *    This program is free software: you can redistribute it and/or  modify
+ *    it under the terms of the GNU Affero General Public License, version 3,
+ *    as published by the Free Software Foundation.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+'use strict';
+
+const log = require('../net2/logger.js')(__filename);
+const Sensor = require('./Sensor.js').Sensor;
+const HostManager = require("../net2/HostManager.js");
+const hostManager = new HostManager("cli", 'client', 'info');
+const DNSMASQ = require('../extension/dnsmasq/dnsmasq.js');
+const dnsmasq = new DNSMASQ();
+const featureName = 'local_domain';
+const f = require('../net2/Firewalla.js');
+const FILTER_DIR = f.getUserConfigFolder() + "/dnsmasq";
+const LOCAL_DEVICE_DOMAIN = FILTER_DIR + "/local_device_domain.conf";
+const util = require('util');
+const fs = require('fs');
+const unlinkAsync = util.promisify(fs.unlink);
+class LocalDomainSensor extends Sensor {
+    constructor() {
+        super();
+    }
+    run() {
+        this.hookFeature(featureName);
+    }
+    async globalOn() {
+        const hosts = await hostManager.getHostsAsync();
```

Comment:
maybe only last-7day online devices should be considered to support local domain, this can reduce some file update effort

Reply:
`getHostsAsync ` is to return the online devices of the last 7 days.